### PR TITLE
OCPBUGS-3990: Add HyperShift specific priorityClass

### PR DESCRIPTION
**Pull request by jsafrane**
HyperShift needs guest cluster control plane components with priority `"hypershift-control-plane"`, https://hypershift-docs.netlify.app/how-to/distribute-hosted-cluster-workloads/

@openshift/storage

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -654,3 +657,14 @@ func withAWSRegion(infraLister v1.InfrastructureLister) dc.DeploymentHookFunc {
...
+func withHypershiftPriorityClass(isHypershift bool) dc.DeploymentHookFunc {
+	if !isHypershift {
+		return nil
+	}
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		deployment.Spec.Template.Spec.PriorityClassName = hypershiftPriorityClass
+		return nil
+	}
+
+}
```
Comment (bertinatto) [inline pkg/operator/starter.go:670]: IMO the `withHypershiftDeploymentHook` function would be a better place for this, what do you think?

Comment (jsafrane) [reply pkg/operator/starter.go:670]: Agreed.

Initial, I saw a separate `withHypershiftReplicasHook`, so it looks like the idea was to have fine-grained hooks. But it looks weird to have them *that* fine grained.

Comment (jsafrane) [reply pkg/operator/starter.go]: Fixed & pushed

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -169,6 +171,7 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		controlPlaneInformersForEvents,
 		withHypershiftDeploymentHook(isHypershift, os.Getenv(hypershiftImageEnvName)),
 		withHypershiftReplicasHook(isHypershift, guestNodeInformer.Lister()),
+		withHypershiftPriorityClass(isHypershift),
```
Comment (bertinatto) [inline pkg/operator/starter.go:174]: You forgot to remove this line

Comment (jsafrane) [reply pkg/operator/starter.go]: oops, fixed

**Pull request merged**